command/run.go

```diff
@@ -31,11 +31,22 @@ Examples:
 	2. Read commands from standard input and execute in parallel.
 		 > cat commands.txt | s5cmd {{.HelpName}}
 `
+var runCommandFlags = []cli.Flag{
+	&cli.StringFlag{
+		Name:  "source-region",
+		Usage: "[default] connect to a specific region of the remote object storage service",
+	},
+	&cli.StringFlag{
+		Name:  "region",
+		Usage: "[default] region of the destination bucket for cp/mv operations; defaulted to source-region",
+	},
+}
 
 var runCommand = &cli.Command{
 	Name:               "run",
 	HelpName:           "run",
 	Usage:              "run commands in batch",
+	Flags:              runCommandFlags, // to be passed to children subcommands
```

Comment:
Have you tried these flags for run command? I suspect it won't work because the context of run command is actually not the parent context of other subcommands. We parse the given file and create a new context with new flagset ([ref](https://github.com/peak/s5cmd/blob/master/command/run.go#L121)).  You might not be able to access through
```
c.String("region")
c.String("source-region")
```
Can you try and confirm that it works as expected?

Reply:
Yeah, it seems `run` command flags do get passed into subcommands. I did some manual tests using command line with aws s3. Ex:

 -  go run main.go run commands.txt

ERROR "ls s3://bucket-test-3/": AuthorizationHeaderMalformed: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2' status code: 400, request id: FED09052D9A38085, host id: Tb8h/jUhVtBZd/Yr1OAMt316k4RE8p37sjG54JfMeQ+tqXNtfsiFfbwcLmSlStRbwoL8lcd7rC4=

- go run main.go run --source-region=us-west-2 commands.txt

    2020/07/28 12:53:38              1082 LICENSE
2020/07/18 11:50:25               938 go.mod

Reply:
sorry, you were right. flags do get transferred to child contexts from parents, but if same flags are used they do not.